### PR TITLE
Feat: Multiple tokens form

### LINKED ISSUE #16: [THL] Ability to add multiple tokens and a threshold for each
As a User, I want to define multiple tokens and thresholds. 

I also want to define if I want the accounts to meet **any condition or all conditions**.

Acceptance criteria:
- There is a way to add a new entry
- There is a way to remove an entry until it's the only entry
- There is a toggle before the list (https://ui.shadcn.com/docs/components/switch) to toggle between "Account must meet all conditions" and "Account must meet at least one condition")

**For the toggle** 
Let's consider two conditions:

Condition 1: The user must possess a minimum of 5 tokens of token A.
Condition 2: The user must possess a minimum of 3 tokens of token B.

Now, suppose that Condition 1 is satisfied by:
Account 1, Account 2

And Condition 2 is satisfied by:
Account 2, Account 3

In the case of the AND statement, the resulting list should only include:
Account 2 (as it satisfies both conditions).

For the OR statement, the resulting list should include:
Account 1, Account 2, Account 3 (since all accounts satisfy at least one condition).

Image

### PR DESCRIPTION
**Pull request by StanislawMazowieckiAriane**
Fixes #16

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -41,22 +43,50 @@ const App = () => {
...
+  };
 
+  const fetchAllData = async () => {
+    try {
+      const responses = await Promise.all(
+        formData.map(async (item) => {
+          const tokenId = item.tokenId;
+          const minAmount = item.minAmount;
+          const url = `${nodeUrl}/api/v1/tokens/${tokenId}/balances?account.balance=gte:${minAmount}&limit=100`;
+          return fetchData(url);
```
Comment (michalrozekariane) [inline src/App.tsx]: I think you can destructure `tokenId` and `minAmount` here right away

**Review by michalrozekariane — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (StanislawMazowieckiAriane) [reply src/App.tsx]: Done

File: src/App.tsx [hunk 2]
```diff
@@ -53,22 +55,48 @@ const App = () => {
...
+    if (isAllConditionsRequired) {
+      return data.filter(
+        (balance, index, self) =>
+          self.findIndex((b) => b.account === balance.account) === index &&
+          responses.every((response) => response.some((b) => b.account === balance.account)),
+      );
+    } else {
+      return data.filter((balance, index, self) => self.findIndex((b) => b.account === balance.account) === index);
+    }
+  };
```
Comment (piotrswierzy) [inline src/App.tsx:70]: Maybe write a unit test for it

**Review by piotrswierzy — CHANGES_REQUESTED** [review]

**Comment by StanislawMazowieckiAriane** [comment]
This PR is too outdated regarding https://github.com/Ashe-Oro-Accelerator/token-holders-list/pull/18
It contains several bugs etc which are fixed in the new PR.
Closing

**Comment by StanislawMazowieckiAriane** [comment]
This PR is too outdated regarding https://github.com/Ashe-Oro-Accelerator/token-holders-list/pull/18
It contains several bugs etc which are fixed in the new PR.
Closing

**Pull request closed without merging**